java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/ArrowFlightConnection.java

```diff
@@ -147,6 +152,7 @@ synchronized ExecutorService getExecutorService() {
         executorService;
   }
 
+  @SuppressFBWarnings(value = "EI_EXPOSE_REP", justification = "We shouldn't make copies of Properties")
```

Comment:
EI: May expose internal representation by returning reference to mutable object (EI_EXPOSE_REP)
Returning a reference to a mutable object value stored in one of the object's fields exposes the internal representation of the object.  If instances are accessed by untrusted code, and unchecked changes to the mutable object would compromise security or other important properties, you will need to do something different. Returning a new copy of the object is better approach in many situations.

Reply:
Do we need to return a mutable reference to the properties though?
Should we instead of exposing the Properties object itself, add getProperty() function? Should we return an immutable view over the Properties?